tools/diffscripts/diffapi.ps1

```diff
@@ -0,0 +1,203 @@
+
+
+$internalroot = Join-Path -Path $baseroot -Child "\iot-sdks-internals"
+
+if ($AsmToolPath -ne $null) {
+    $internalroot = $SDKInternalsPath
+}
+
+$compareDirectory = Join-Path -Path $internalroot -Child "\sdk_design_docs\CSharp\master"
+
+if ($Preview) {
+    $compareDirectory = Join-Path -Path $internalroot -Child "\sdk_design_docs\CSharp\preview"
+}
+
+# Hardcoded list of assembly names
+$filenames = @(
+    "Microsoft.Azure.Devices.Shared",
+    "Microsoft.Azure.Devices.Client",
+    "Microsoft.Azure.Devices",
+    "Microsoft.Azure.Devices.Provisioning.Client",
+    "Microsoft.Azure.Devices.Provisioning.Service",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Amqp",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Mqtt",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Http",
+    "Microsoft.Azure.Devices.Provisioning.Security.Tpm"
+)
+
+# All of the files from the build
+$oldsetNames = @(
+    ((Join-Path -Path $reporoot -ChildPath "\shared\src\bin\Release\netstandard2.1\")+($filenames[0]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\iothub\device\src\bin\Release\netstandard2.1\")+($filenames[1]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\iothub\service\src\bin\Release\netstandard2.1\")+($filenames[2]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\device\src\bin\Release\netstandard2.1\")+($filenames[3]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\service\src\bin\Release\netstandard2.1\")+($filenames[4]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\transport\amqp\src\bin\Release\netstandard2.1\")+($filenames[5]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\transport\mqtt\src\bin\Release\netstandard2.1\")+($filenames[6]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\transport\http\src\bin\Release\netstandard2.1\")+($filenames[7]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\security\tpm\src\bin\Release\netstandard2.1\")+($filenames[8]+".dll"))
+)
+
+# Create a list of the markdown files so we can compare them to the API doc directory
+$markdownoutput = @(
+    ($compareDirectory+ "\" + ($filenames[0]+".md")),
+    ($compareDirectory+ "\" + ($filenames[1]+".md")),
+    ($compareDirectory+ "\" + ($filenames[2]+".md")),
+    ($compareDirectory+ "\" + ($filenames[3]+".md")),
+    ($compareDirectory+ "\" + ($filenames[4]+".md")),
+    ($compareDirectory+ "\" + ($filenames[5]+".md")),
+    ($compareDirectory+ "\" + ($filenames[6]+".md")),
+    ($compareDirectory+ "\" + ($filenames[7]+".md")),
+    ($compareDirectory+ "\" + ($filenames[8]+".md"))
+)
+
+if ((Test-Path $compareDirectory) -ne $TRUE) {
+    Write-Host "You have not cloned the sdk internals repository. Or it is not in the same root repository collection location."
+    Write-Host "For example, if this SDK is cloned to c:\repos you should clone https://github.com/Azure/iot-sdks-internals.git to the c:\repos folder."
```

Comment:
Why have this requirement of all repos sharing a common root?

Reply:
It's how I do it so it made it easier to do this. It's not a requirement just a convenience. I'll add a note to specify the -AsmToolPath as well

Reply:
FWIW, mine are all under c:\code, and then I have a second level of directories for where the repo is from, like `azure` or `azure-samples` or `drwill-ms` (for my forks).